src/Bridges/bridgeoptimizer.jl

```diff
@@ -209,25 +209,59 @@ function MOI.addconstraint!(b::AbstractBridgeOptimizer, f::MOI.AbstractFunction,
         MOI.addconstraint!(b.model, f, s)
     end
 end
-function MOI.canmodifyconstraint(b::AbstractBridgeOptimizer, ci::CI, change)
-    if isbridged(b, typeof(ci))
-       MOI.canmodifyconstraint(b.bridged, ci, change) && MOI.canmodifyconstraint(b, MOIB.bridge(b, ci), change)
+function MOI.canmodify(b::AbstractBridgeOptimizer, ::Type{CI{F, S}}, ::Type{Chg}) where {F, S, Chg<:MOI.AbstractFunctionModification}
+    if isbridged(b, CI{F, S})
+        MOI.canmodify(b.bridged, CI{F, S}, Chg) && MOI.canmodify(b, MOIB.bridgetype(b, F, S), Chg)
     else
-        MOI.canmodifyconstraint(b.model, ci, change)
+        MOI.canmodify(b.model, CI{F, S}, Chg)
     end
 end
-function MOI.modifyconstraint!(b::AbstractBridgeOptimizer, ci::CI, change)
+function MOI.modify!(b::AbstractBridgeOptimizer, ci::CI, change)
     if isbridged(b, typeof(ci))
-        MOI.modifyconstraint!(b, bridge(b, ci), change)
-        MOI.modifyconstraint!(b.bridged, ci, change)
+        MOI.modify!(b, bridge(b, ci), change)
+        MOI.modify!(b.bridged, ci, change)
+    else
+        MOI.modify!(b.model, ci, change)
+    end
+end
+
+function MOI.canset(b::AbstractBridgeOptimizer, attr::MOI.ConstraintSet, ::Type{CI{F, S}}) where {F, S}
+    if isbridged(b, CI{F, S})
+        MOI.canset(b.bridged, MOI.ConstraintSet(), CI{F, S}) && MOI.canset(b, MOI.ConstraintSet(), MOIB.bridgetype(b, F, S))
+    else
+        MOI.canset(b.model, MOI.ConstraintSet(), CI{F, S})
+    end
+end
+function MOI.set!(b::AbstractBridgeOptimizer, ::MOI.ConstraintSet, constraint_index::CI, set)
+    # Note: we also under-type this function to avoid ambiguity
+    if isbridged(b, typeof(constraint_index))
+        MOI.set!(b, MOI.ConstraintSet(), bridge(b, constraint_index), set)
+        MOI.set!(b.bridged, MOI.ConstraintSet(), constraint_index, set)
+    else
+        MOI.set!(b.model, MOI.ConstraintSet(), constraint_index, set)
+    end
+end
+
+function MOI.canset(b::AbstractBridgeOptimizer, ::MOI.ConstraintFunction, ::Type{CI{F, S}}) where {F, S}
+    if isbridged(b, CI{F, S})
+        MOI.canset(b.bridged, MOI.ConstraintFunction(), CI{F, S}) && MOI.canset(b, MOI.ConstraintFunction(), MOIB.bridgetype(b, F, S))
+    else
+        MOI.canset(b.model, MOI.ConstraintFunction(), CI{F, S})
+    end
+end
+function MOI.set!(b::AbstractBridgeOptimizer, ::MOI.ConstraintFunction, constraint_index::CI, func)
+    # Note: we also under-type this function to avoid ambiguity
```

Comment:
What do you mean by this comment ? Why don't you just define `set!(..., attr::Union{ConstraintFunction, ConstraintSet}, ...)` ?

Reply:
I had a hard time getting rid of method ambiguities since there are quite a few abstract subtypes of ModelLike floating around. With the Union there is an ambiguity with `set!(::ModelLike, ::ConstraintFunction, ...)`. as the first argument is more specific but the second is less specific. I've typed this now and it seems to all work.